### PR TITLE
privilege: fix privilege check of `GRANT ROLE`

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
In MySQL, `GRANT roles TO users` require `ROLE_ADMIN` or `SUPER` privilege, But TiDB use `GrantPriv` to check `GRANT ROLE`.

### What is changed and how it works?
fix this bug and add test.

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

Release note

 - fix privilege check rule for `GRANT ROLE`

### CODE REVIEW

File: executor/simple_test.go [hunk 1]
```diff
@@ -188,6 +188,28 @@ func (s *testSuite3) TestRole(c *C) {
...
+	se, err := session.CreateSession4Test(s.store)
+	c.Check(err, IsNil)
+	defer se.Close()
+	c.Assert(se.Auth(&auth.UserIdentity{Username: "testRoleAdmin", Hostname: "localhost"}, nil, nil), IsTrue)
+
+	ctx := context.Background()
+	_, err = se.Execute(ctx, "GRANT `targetRole` TO `testRoleAdmin`;")
+	c.Assert(err, IsNil)
+	_, err = se.Execute(ctx, "REVOKE `targetRole` FROM `testRoleAdmin`;")
+	c.Assert(err, IsNil)
```
Comment (lamxTyler) [inline executor/simple_test.go]: Do we need a fail case like only have `grant_role` access?

Comment (imtbkcat) [reply executor/simple_test.go]: A failed case added, PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**